### ISSUE TITLE
一个评估组件连多个算法board 界面仅显示其中某个算法

### ISSUE THREAD

**Issue by jennyjue** [FederatedAI/FATE#2421]
最终board界面：
![image](https://user-images.githubusercontent.com/71378193/103056560-3a66cc00-45d8-11eb-89bf-d315a9fd8c73.png)
![image](https://user-images.githubusercontent.com/71378193/103056573-43579d80-45d8-11eb-8108-af58ff80ad2d.png)

dsl如下：
{
    "components" : {
        "reader_0": {
            "module": "Reader",
            "output": {
                "data": ["table"]
            }
        },
        "dataio_0": {
            "module": "DataIO",
            "input": {
                "data": {
                    "data": [
                        "reader_0.table"
                    ]
                }
            },
            "output": {
                "data": ["train"],
                "model": ["dataio"]
            }
         },
        "intersection_0": {
            "module": "Intersection",
            "input": {
                "data": {
                    "data": [
                        "dataio_0.train"
                    ]
                }
            },
            "output": {
                "data": ["train"]
           }
        },
        "federated_sample_0": {
            "module": "FederatedSample",
            "input": {
                "data": {
                    "data": [
                        "intersection_0.train"
                    ]
                }
            },
            "output": {
                "data": ["train"]
            }
        },
        "feature_scale_0": {
            "module": "FeatureScale",
            "input": {
                "data": {
                    "data": [
                        "federated_sample_0.train"
                    ]
                }
            },
            "output": {
                "data": ["train"],
                "model": ["feature_scale"]
            }
        },
        "hetero_feature_binning_0": {
            "module": "HeteroFeatureBinning",
            "input": {
                "data": {
                    "data": [
                        "feature_scale_0.train"
                    ]
                }
            },
            "output": {
                "data": ["train"],
                "model": ["hetero_feature_binning"]
            }
        },
        "hetero_feature_selection_0": {
            "module": "HeteroFeatureSelection",
            "input": {
                "data": {
                    "data": [
                        "hetero_feature_binning_0.train"
                    ]
                },
                "isometric_model": [
                    "hetero_feature_binning_0.hetero_feature_binning"
                ]
            },
            "output": {
                "data": ["train"],
                "model": ["selected"]
            }
        },
        "one_hot_0": {
            "module": "OneHotEncoder",
            "input": {
                "data": {
                    "data": [
                        "hetero_feature_selection_0.train"
                    ]
                }
            },
            "output": {
                "data": ["train"],
                "model": ["onehot_encoder"]
            }
        },
        "hetero_lr_0": {
            "module": "HeteroLR",
            "input": {
                "data": {
                    "train_data": ["one_hot_0.train"]
                }
            },
            "output": {
                "data": ["train"],
                "model": ["hetero_lr"]
            },
            "need_deploy": false
        },
        "hetero_lr_1": {
            "module": "HeteroLR",
            "input": {
                "data": {
                    "train_data": ["one_hot_0.train"]
                }
            },
            "output": {
                "data": ["train"]
            },
            "need_deploy": false
        },
        "hetero_secureboost_0": {
            "module": "HeteroSecureBoost",
            "input": {
                "data": {
                    "train_data": ["one_hot_0.train"]
                }
            },
            "output": {
                "data": ["train"]
            },
            "need_deploy": false
        },
        "hetero_secureboost_1": {
            "module": "HeteroSecureBoost",
            "input": {
                "data": {
                    "train_data": ["one_hot_0.train"]
                }
            },
            "output": {
                "data": ["train"],
                "model": ["train"]
            }
        },
        "evaluation_0": {
            "module": "Evaluation",
            "input": {
                "data": {
                    "data": [
                        "hetero_lr_0.train", "hetero_secureboost_0.train","hetero_lr_1.train", "hetero_secureboost_1.train"
                    ]
                }
            }
        }
    }
}

conf如下：
{
    "dsl_version": "2",
    "initiator": {
        "role": "guest",
        "party_id": 9999
    },
    "role": {
        "guest": [
            9999
        ],
        "host": [
            10000
        ],
        "arbiter": [
            10000
        ]
    },
    "job_parameters": {
        "common": {
            "work_mode": 1,
            "task_parallelism": 2,
            "computing_partitions": 8,
            "eggroll_run": {
                "eggroll.session.processors.per.node": 2
            },
            "spark_run": {
                "num-executors": 1,
                "executor-cores": 2
            }
        }
    },
    "component_parameters": {
        "common": {
            "federated_sample_0": {
                "mode": "stratified",
                "method": "upsample",
                "fractions": [
                    [
                        0,
                        1.5
                    ],
                    [
                        1,
                        2.0
                    ]
                ]
            },
            "hetero_lr_0": {
                "penalty": "L2",
                "optimizer": "rmsprop",
                "alpha": 0.01,
                "max_iter": 3,
                "batch_size": 320,
                "learning_rate": 0.15,
                "init_param": {
                    "init_method": "random_uniform"
                }
            },
            "hetero_lr_1": {
                "penalty": "L2",
                "optimizer": "rmsprop",
                "alpha": 0.01,
                "max_iter": 3,
                "batch_size": 320,
                "learning_rate": 0.15,
                "init_param": {
                    "init_method": "random_uniform"
                },
                "cv_param": {
                    "n_splits": 2,
                    "shuffle": true,
                    "random_seed": 103,
                    "need_cv": true
                }
            },
            "hetero_secureboost_0": {
                "cv_param": {
                    "n_splits": 2,
                    "need_cv": true,
                    "shuffle": false
                },
                "num_trees": 3
            },
            "hetero_secureboost_1": {
                "num_trees": 2
            }
        },
        "role": {
            "guest": {
                "0": {
                    "reader_0": {
                        "table": {"name": "breast_hetero_guest", "namespace": "experiment"}
                    },
                    "dataio_0":{
                        "with_label": true,
                        "label_name": "y",
                        "label_type": "int",
                        "output_format": "dense",
                        "missing_fill": true,
                        "outlier_replace": true
                    }
                }
            },
            "host": {
                "0": {
                    "reader_0": {
                        "table": {"name": "breast_hetero_host", "namespace": "experiment"}
                    },
                    "dataio_0":{
                        "with_label": false,
                        "output_format": "dense",
                        "outlier_replace": true
                    }
                }
            }
        }
    }
}
